spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/DatastoreTemplate.java

```diff
@@ -523,7 +527,7 @@ private static StructuredQuery.OrderBy createOrderBy(
 
   private List<Entity> convertToEntityForSave(
       Object entity, Set<Key> persistedEntities, Key... ancestors) {
-    if (ancestors != null) {
+    if (ancestors.length != 0) {
```

Comment:
Initially I thought that since `varargs` are treated as arrays, checking length should be enough. In our case, when there is no ancestors, no argument of type Key is passed down, and `ancestors` would be an empty array.
But since you point this out, I found if only a `null` is passed in, then `ancestors == null`. So I'll add both checks just to be cautious.